docs/io/fits/index.rst

```diff
@@ -168,11 +168,12 @@ Working with compressed files
 The :func:`open` function will seamlessly open FITS files that have been
 compressed with gzip, bzip2 or pkzip. Note that in this context we are talking
 about a FITS file that has been compressed with one of these utilities (e.g., a
-.fits.gz file).
+.fits.gz file). Opening bzip2-compressed files requires the :mod:`bz2` module
+to be available.
 
 There are some limitations when working with compressed files. For example,
 with Zip files that contain multiple compressed files, only the first file will
-be accessible. Also bzip does not support the append or update access modes.
+be accessible. Also bzip2 does not support the append or update access modes.
```

Comment:
The change is reasonable. However it seems the sentence itself is outdated. When I look on the [module docs](https://docs.python.org/3/library/bz2.html#bz2.BZ2File) it seems like `bz2` supports appending and overwriting. I will add a new issue for this.

Reply:
That would be very helpful of course, then at least [file.py](https://github.com/astropy/astropy/blob/de2be96c0b3aa1b353b872e8e97de99b1484d1b1/astropy/io/fits/file.py#L462) and probably a couple of tests will need to be updated. Not sure if `get_readable_fileobj` already supports this. `file.py` currently also rejects appending to `gzip` files; if that still holds it should probably also be mentioned in the docs. Seems a case for a new PR.

Reply:
Of course that should be done seperately. I just wanted to mention it so it's not forgotten.

Reply:
See #9685